test/test-metrics.c

```diff
@@ -0,0 +1,112 @@
+#include "uv.h"
+#include "task.h"
+#include <string.h> /* memset */
+
+
+typedef struct {
+  int metrics_run_count;
+  int timer_run_count;
+} mtc_t;
+
+typedef struct {
+  uv_async_t async;
+  uv_loop_t mloop;
+  uv_timer_t timer;
+  uv_thread_t thread;
+  uv_timer_t ttimer;
+  int thread_timer_count;
+} mthread_t;
+
+
+static void timer_noop_cb(uv_timer_t* handle) {
+}
+
+TEST_IMPL(metrics_idle_time) {
+  uv_timer_t timer;
+  int r;
+  uint64_t pt;
+
+  uv_loop_configure(uv_default_loop(), UV_LOOP_IDLE_TIME);
+  uv_timer_init(uv_default_loop(), &timer);
+  uv_timer_start(&timer, timer_noop_cb, 100, 0);
+
+  r = uv_run(uv_default_loop(), UV_RUN_DEFAULT);
+  ASSERT(r == 0);
+
+  pt = uv_metrics_idle_time(uv_default_loop());
+
+  /* Technically should be able to check if >= 100 ms, but the timer might run
+   * slightly early. So check if >= expected time - 1 ms.
+   */
```

Comment:
Nit: may check that the timer cb has been called.

Reply:
Good call.